dali/pipeline/operators/geometric/flip.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATORS_GEOMETRIC_FLIP_H_
+#define DALI_PIPELINE_OPERATORS_GEOMETRIC_FLIP_H_
+
+#include <vector>
+#include <string>
+#include "dali/kernels/tensor_shape.h"
+#include "dali/pipeline/data/backend.h"
+#include "dali/pipeline/operators/operator.h"
+
+namespace dali {
+
+template <typename Backend>
+class Flip: public Operator<Backend> {
+ public:
+  explicit Flip(const OpSpec &spec);
+
+  ~Flip() override = default;
+  DISABLE_COPY_MOVE_ASSIGN(Flip);
+
+ protected:
+  void RunImpl(Workspace<Backend> *ws, const int idx) override;
+
+  bool GetHorizontal(const ArgumentWorkspace *ws, int idx) {
+    return this->spec_.template GetArgument<int32>("horizontal", ws, idx);
+  }
+
+  bool GetVertical(const ArgumentWorkspace *ws, int idx) {
+    return this->spec_.template GetArgument<int32>("vertical", ws, idx);
+  }
+
+  std::vector<int32> GetHorizontal(const ArgumentWorkspace *ws) {
+    return GetTensorArgument(ws, "horizontal");
```

Comment:
In fact, it is better to make returned types consistent, although a proper type here is `int`, because it is how the `horizontal` and `vertical` arguments are defined.